From 4bb1d3e17220b903b102d64a62fdb04f2bffa88a Mon Sep 17 00:00:00 2001
From: Avery Ching <aching@fb.com>
Date: Mon, 27 Oct 2014 16:12:01 -0700
Subject: [PATCH] GIRAPH-960: Max task attempts should be 1 and not 0.

Summary: 0 is an invalid number that may not work on some Hadoop distributions

Test Plan: Ran a job and failed a task.

Reviewers: pavanka.26, maja.kabiljo

Reviewed By: maja.kabiljo

Differential Revision: https://reviews.facebook.net/D27579
---
 .../src/main/java/org/apache/giraph/job/GiraphJob.java      | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java b/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java
index 11ae7fc34..79a145fa1 100644
--- a/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java
+++ b/giraph-core/src/main/java/org/apache/giraph/job/GiraphJob.java
@@ -218,14 +218,14 @@ public final boolean run(boolean verbose)
     giraphConfiguration.setBoolean("mapreduce.job.user.classpath.first", true);
 
     // If the checkpoint frequency is 0 (no failure handling), set the max
-    // tasks attempts to be 0 to encourage faster failure of unrecoverable jobs
+    // tasks attempts to be 1 to encourage faster failure of unrecoverable jobs
     if (giraphConfiguration.getCheckpointFrequency() == 0) {
       int oldMaxTaskAttempts = giraphConfiguration.getMaxTaskAttempts();
-      giraphConfiguration.setMaxTaskAttempts(0);
+      giraphConfiguration.setMaxTaskAttempts(1);
       if (LOG.isInfoEnabled()) {
         LOG.info("run: Since checkpointing is disabled (default), " +
             "do not allow any task retries (setting " +
-            GiraphConstants.MAX_TASK_ATTEMPTS.getKey() + " = 0, " +
+            GiraphConstants.MAX_TASK_ATTEMPTS.getKey() + " = 1, " +
             "old value = " + oldMaxTaskAttempts + ")");
       }
     }
